### DISCUSSION TITLE
Support internal interface setters

### DISCUSSION THREAD

**Discussion by nameOfUser0236** [dotnet/csharplang#5459] [Language Ideas] [answered]
The following code produces this error:
`'MyClass' does not implement interface member 'IMyInterface.MyNumber.set'. 'MyClass.MyNumber.set' is not public.`

```c#
public interface IMyInterface
{
	public int MyNumber { get; internal set;}
}
public class MyClass : IMyInterface
{
	public int MyNumber { get; internal set;}
}
```
Would you be so kind as to remove this restriction?
It has been greatly aggravating.

**Comment by svick** [comment, marked as answer]
I don't know if the current behavior is expected, but [it is possible to implement that interface explicitly](https://sharplab.io/#v2:C4LgbgpgTglgZgTwDQBMQGoA+ABAzAAhgDtho4BDAYwnwEkBZBWksqiAWACgBvLgSDyES+RgDkArgFsARtHzd8AcwjAA3ENJQi5ADb4AzivUBfLqc6DsAJhEIAwjvL79+EHUbNNFal16cBBMTAthIycgrKahrQ2nqGUeb4SRruTCxQ3hAAdGJSslDySir4ALwAfCF50OrxpRW5YQUl+GC64hAmZkA===):

```c#
public class MyClass : IMyInterface
{
    public int MyNumber { get; internal set; }
    int IMyInterface.MyNumber { get => MyNumber; set => MyNumber = value; }
}
```

**Reply by nameOfUser0236** [reply]
Thank you for helping me solve my current problem!!! However, I hope, in the future, that this limitation will be removed.
